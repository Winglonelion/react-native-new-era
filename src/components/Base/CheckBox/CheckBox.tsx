import React, { memo } from 'react';
import { ColorValue, StyleSheet, View } from 'react-native';
import Colors from 'theme/colors';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';

const SUPPORTED_ICON = {
  Feather,
  FontAwesome5,
};

interface PropTypes {
  checked?: boolean;
  tintColor?: ColorValue;
  checkedIconName?: string;
  iconColor?: string;
  iconSet?: 'Feather' | 'FontAwesome5';
}

const CheckBox: React.FC<PropTypes> = memo(
  ({
    checked = false,
    tintColor = Colors.orange,
    checkedIconName = 'check',
    iconColor = Colors.white,
    iconSet = 'FontAwesome5',
  }) => {
    const Icon = SUPPORTED_ICON[iconSet];
    return (
      <View
        style={[
          styles.container,
          { backgroundColor: checked ? tintColor : Colors.uncheck },
        ]}>
        {checked ? (
          <Icon name={checkedIconName} size={14} color={iconColor} />
        ) : (
          <View style={styles.unCheckView} />
        )}
      </View>
    );
  },
);

const styles = StyleSheet.create({
  container: {
    width: 20,
    height: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  unCheckView: {
    width: 16,
    height: 16,
    backgroundColor: Colors.white,
  },
});

export default CheckBox;
