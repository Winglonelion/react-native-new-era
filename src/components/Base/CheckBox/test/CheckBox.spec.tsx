import React from 'react';
import { shallow } from 'enzyme';
import CheckBox, { PropTypes } from '../CheckBox';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<CheckBox {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('CheckBox check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('checked should match snapshot', () => {
    const { wrapper } = setup({ checked: true });
    expect(wrapper).toMatchSnapshot();
  });
});
describe('CheckBox check render correctly', () => {
  test('unchecked should contain 2 View', () => {
    const { wrapper } = setup();
    expect(wrapper.find('View')).toHaveLength(2);
  });

  test('checked should contain 1 View, 1 Icon', () => {
    const { wrapper } = setup({ checked: true });
    expect(wrapper.find('IconClass')).toHaveLength(1);
    expect(wrapper.find('View')).toHaveLength(1);
  });
});
