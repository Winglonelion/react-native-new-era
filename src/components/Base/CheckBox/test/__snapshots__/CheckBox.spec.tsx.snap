// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox check snapshot checked should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      },
      Object {
        "backgroundColor": "#FF894D",
      },
    ]
  }
>
  <IconClass
    brand={false}
    color="#FFFFFF"
    light={false}
    name="check"
    regular={false}
    size={14}
    solid={false}
  />
</View>
`;

exports[`CheckBox check snapshot should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      },
      Object {
        "backgroundColor": "#C4C1C0",
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "height": 16,
        "width": 16,
      }
    }
  />
</View>
`;
